eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -1038,17 +1038,48 @@ public V removeKey(K key)
     @Override
     public boolean removeIf(Predicate2<? super K, ? super V> predicate)
     {
-        int previousSize = this.size();
-        Iterator<Entry<K, V>> iterator = this.entrySet().iterator();
-        while (iterator.hasNext())
+        int previousOccupied = this.occupied;
```

Comment:
I'm curious about `occupied` vs `size()`. Is there an advantage to one over the other?

Reply:
size() is just a wrapper around occupied, so though of using the field directly.